rueidiscompat/adapter.go

```diff
@@ -1093,14 +1093,36 @@ func (c *Compat) SetBit(ctx context.Context, key string, offset int64, value int
 	return newIntCmd(resp)
 }
 
+const (
+	BitCountIndexByte = "BYTE"
+	BitCountIndexBit  = "BIT"
+)
+
 func (c *Compat) BitCount(ctx context.Context, key string, bitCount *BitCount) *IntCmd {
-	var resp rueidis.RedisResult
+	ret := &IntCmd{}
+
 	if bitCount == nil {
-		resp = c.client.Do(ctx, c.client.B().Bitcount().Key(key).Build())
-	} else {
-		resp = c.client.Do(ctx, c.client.B().Bitcount().Key(key).Start(bitCount.Start).End(bitCount.End).Build())
+		resp := c.client.Do(ctx, c.client.B().Bitcount().Key(key).Bit().Build()) // Default to BIT
```

Comment:
Hi @SoulPancake,

We can't use .Bit() as default because it is only supported on Redis >= 7. We should keep the original behavior.